src/main/java/org/springframework/integration/dsl/IntegrationFlowDefinition.java

```diff
@@ -2330,11 +2331,11 @@ public B route(MessageProcessorSpec<?> messageProcessorSpec,
 					hasSubFlows = true;
 					IntegrationFlowDefinition<?> flowBuilder = (IntegrationFlowDefinition<?>) component;
 					addComponent(flowBuilder.fixedSubscriberChannel()
```

Comment:
This seems a bit strange - to unconditionally add a channel at the end of the subflow - the subflow might not produce a result. AFAICT, we end up adding a disjoint channel at the end which will never receive any data.